images/win/scripts/Installers/Configure-Shell.ps1

```diff
@@ -0,0 +1,19 @@
+# Create shells folder
+$shellPath = "C:\shells"
+New-Item -Path $shellPath -ItemType Directory | Out-Null
+
+# sh and bash <--> C:\msys64\usr\bin\bash.exe
+New-Item -ItemType SymbolicLink -Path "$shellPath\bash.exe" -Target "C:\msys64\usr\bin\bash.exe" | Out-Null
+New-Item -ItemType SymbolicLink -Path "$shellPath\sh.exe" -Target "C:\msys64\usr\bin\sh.exe" | Out-Null
+
+# winbash <--> C:\Windows\System32\bash.exe
+New-Item -ItemType SymbolicLink -Path "$shellPath\winbash.exe" -Target "$env:SystemRoot\System32\bash.exe" | Out-Null
+
+# git4bash <--> C:\Program Files\Git\bin\bash.exe
+New-Item -ItemType SymbolicLink -Path "$shellPath\git4bash.exe" -Target "$env:ProgramFiles\Git\bin\bash.exe" | Out-Null
+
+# MSYS2 <--> C:\msys64\usr\bin\bash.exe
+New-Item -ItemType SymbolicLink -Path "$shellPath\msys2.exe" -Target "C:\msys64\usr\bin\bash.exe" | Out-Null

```

Comment:
This might break Action setup-msys2. Currently, the Action adds `msys2.cmd` to the PATH, and then `shell: msys2 {0}` is used. I'm afraid that `msys2.exe` might be picked instead of `msys2.cmd` or the other way round. There are several possibilities:

- I might be wrong. It might be that `shell: msys2` corresponds to `msys2.exe` (the symbolic link) and `shell: msys2 {0}` corresponds to `msys2.cmd`.
- Is this symbolic link actually needed? If both `sh` and `bash` do already point to the same target, I feel this to be redundant.
- We can deal with it introducing a breaking change and bumping setup-msys2 to v3, should it be really needed. However, unless the `msys2.exe` symlink is required ATM, I would defer it. I'd say, let's let other changes in this PR stabilize before going forward with "custom" msys2 entrypoint details.

/cc @lazka

Reply:
Fixed. Renamed to msys2bash.exe.